providers/togglz/src/main/java/dev/openfeature/contrib/providers/togglz/TogglzProvider.java

```diff
@@ -0,0 +1,99 @@
+package dev.openfeature.contrib.providers.togglz;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.EventProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.Reason;
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.FlagNotFoundError;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import dev.openfeature.sdk.exceptions.ProviderNotReadyError;
+import dev.openfeature.sdk.exceptions.TypeMismatchError;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.togglz.core.Feature;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Provider implementation for Togglz.
+ */
+@AllArgsConstructor
+@Slf4j
+public class TogglzProvider extends EventProvider {
+
+    @Getter
+    private static final String NAME = "Togglz Provider";
+    public static final String NOT_IMPLEMENTED =
+        "Not implemented - provider does not support this type. Only boolean is supported.";
+
+    private Map<String, Feature> features = new HashMap<>();
+
+    @Getter
+    private ProviderState state = ProviderState.NOT_READY;
+
+    public TogglzProvider(Collection<Feature> featuresCollection) {
+        featuresCollection.forEach(feature -> features.put(feature.name(), feature));
+    }
+
+    /**
+     * Initialize the provider.
+     * @param evaluationContext evaluation context
+     * @throws Exception on error
+     */
+    @Override
+    public void initialize(EvaluationContext evaluationContext) throws Exception {
+        super.initialize(evaluationContext);
+        state = ProviderState.READY;
+        log.debug("finished initializing provider, state: {}", state);
+    }
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> NAME;
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError("provider not yet initialized");
+            }
+            throw new GeneralError("unknown error");
+        }
+        Feature feature = features.get(key);
+        if (feature == null) {
+            throw new FlagNotFoundError("flag " + key + " not found");
+        }
+        boolean featureBooleanValue = feature.isActive();
+        return ProviderEvaluation.<Boolean>builder()
+            .value(featureBooleanValue)
+            .reason(Reason.TARGETING_MATCH.name())
```

Comment:
You may want to use the reason `STATIC` since it doesn't look like evaluation context is being used.

https://openfeature.dev/specification/types#resolution-details

Reply:
Not sure that STATIC is correct here, since [Activation ](https://www.togglz.org/documentation/activation-strategies)can be done by user even without passing context if I understand correctly.
@bennetelli, @hennr can share their thoughts.

Reply:
Wouldn't you need to include `FeatureUser` when calling `isActive`?

Reply:
@beeme1mr I don't think so. It seems like Togglz uses pre-configured features/beans to get these values contextually. For example, to get the user name  you configure one of these: https://www.togglz.org/documentation/authentication, which take the user-name from the servlet context, etc.

This is actually configured in the `feature` (see the constructor) if I understand correctly.